src/k8s/pkg/k8sd/api/cluster_join.go

```diff
@@ -24,42 +22,26 @@ func (e *Endpoints) postClusterJoin(s *state.State, r *http.Request) response.Re
 		return response.BadRequest(fmt.Errorf("invalid hostname %q: %w", req.Name, err))
 	}
 
+	context := r.Context()
```

Comment:
Just use `r.Context()` everywhere. There is no need to reassign this.

Reply:
done!